internal/test/helper.go

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// The OpenSearch Contributors require contributions made to
+// this file be licensed under the Apache-2.0 license or a
+// compatible open source license.
+
+package ostest
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/wI2L/jsondiff"
+
+	"github.com/opensearch-project/opensearch-go/v3"
+	"github.com/opensearch-project/opensearch-go/v3/opensearchapi"
+)
+
+// NewClient returns an opensearchapi.Client that is adjusted for the wanted test case
+func NewClient() (*opensearchapi.Client, error) {
+	if os.Getenv("SECURE_INTEGRATION") == "true" {
```

Comment:
You should be able to refactor this away by having `ClientConfig()` return `nil` when there isn't one defined.